### PR TITLE
Add compute_range method to EchoData, fix #310

### LINKED ISSUE #310: Error when trying to calibrate an EK80 file with both CW and BB channels
Calibration fails for the test file `Summer2018--D20180905-T033113.raw` when the calibration parameters are being retrieved. The error message is: `ValueError: operands could not be broadcast together with shapes (5,5) (3,1)`. 

Specifying the waveform_mode and the encode_mode still resulted in the error.

### PR DESCRIPTION
**Pull request by imranmaj**
Part of #259 

* Adds a `compute_range` method to `EchoData`
    * Changes `CalibrateEk.compute_range_meter` and `CalibrateAZFP.compute_range_meter` to call this new method on `EchoData` internally
    * Adds test for `EchoData.compute_range`
* Removes the `tvg_correction_factor` parameter from `CalibrateEK.compute_range_meter`
* Adds ABC to calibrate_base.py
* Fixes #310 
    * Adds test for file with both CW and BB channels

### CODE REVIEW

File: echopype/tests/test_echodata.py [hunk 1]
```diff
@@ -133,3 +137,33 @@ def _check_path(zarr_path):
...
+        (ad2cp_path / "raw" / "090" / "rawtest.090.00001.ad2cp", "AD2CP", None)
+    ]
+)
+def test_compute_range(filepath, sonar_model, xml_path):
+    ed = echopype.open_raw(filepath, sonar_model, xml_path)
+    sound_speed = 343
+    if sonar_model == "EK60":
+        range = ed.compute_range(sound_speed, ek_waveform_mode="CW")
+    elif sonar_model == "EK80":
+        range = ed.compute_range(sound_speed, ek_waveform_mode="CW")
```
Comment (leewujung) [inline echopype/tests/test_echodata.py:157]: @imranmaj : Is there a test for `ek_waveform_mode="BB"` or is there a reason why that is not included here?

**Review by leewujung — APPROVED** [review]

**Comment by leewujung** [comment]
@imranmaj : thanks, great work! Other than the BB test case above, I think this is ready to go. Let's talk about how to create a function for #259 which is also related to regridding tomorrow.

**Comment by leewujung** [comment]
@imranmaj : Thanks for this great PR with new functionality and under the hood improvements!  😃 

### docstring improvement
I just have one thing for improving the docstring for `compute_range`: While testing `compute_range` for AZFP, I found that it is not obvious to the user that they need to supply `env_params` since sound speed is not recorded in the data and needs to be computed from temperature, salinity, and pressure. And also that salinity and pressure also need to be supplied by the user because they are not part of the data file. 

Could you expand the docstring of `compute_range` in a fashion similar to EK60/80 so that users can have the above info?


### Note for the future
Another thing I wanted to note here is that, ideally we should add a test to run cal on `waveform_mode="BB"` and `encode_mode="complex"` option for the file `Summer2018--D20180905-T033113.raw` (the file with co-existing CW+BB data) -- this test currently is not included. However, the size of the file (305 MB) makes it too long to run it. We'll need to optimize the code for the many convolutions needed in the operation, but I want to note this down here so we do not forget and also get this on @lsetiawan's radar for when we make test data changes for speeding up CI runs.

**Comment by imranmaj** [comment]
The CI is failing because I pulled the dev branch, but the current dev branch has issues with pre-commit so I reverted the pull. It seems like pre-commit won't accept the reversion that I did and permanently gives me an error.

**Comment by leewujung** [comment]
From just the pre-commit.ci messages, it seems that it is a line being too long in combine.py? Is that it or is that a symptom of something else?

**Comment by lsetiawan** [comment]
> From just the pre-commit.ci messages, it seems that it is a line being too long in combine.py? Is that it or is that a symptom of something else?

Seems like that's the only problem I see. Actual python tests are passing.

**Comment by imranmaj** [comment]
Yes, it says that combine.py line 84 is too long, but it's incorrect; in the actual code ([https://github.com/imranmaj/echopype/blob/a83819540c4e1f700428bc46f6a1f6f5dcffb583/echopype/echodata/combine.py#L84](https://github.com/imranmaj/echopype/blob/a83819540c4e1f700428bc46f6a1f6f5dcffb583/echopype/echodata/combine.py#L84)), the line is only 97 characters long and it was not modified by this PR.

**Comment by lsetiawan** [comment]
I see 2 force pushes happened. I feel like pre-commit is confused as to what to check?

**Comment by lsetiawan** [comment]
I think it's safe to just merge it since I just checked locally and it passed.

**Comment by imranmaj** [comment]
> I see 2 force pushes happened.

Yes, I pulled the latest dev branch but the latest dev branch has the pre-commit issue. I fixed the issue in #425 ([https://github.com/OSOceanAcoustics/echopype/pull/425/files#diff-cce6b1b1f93becbc188e01b21cf0a711ad216692eea7e96f98642ebc4579059fR84-R85](https://github.com/OSOceanAcoustics/echopype/pull/425/files#diff-cce6b1b1f93becbc188e01b21cf0a711ad216692eea7e96f98642ebc4579059fR84-R85)) but I didn't want to duplicate the fix, so I tried to revert it.

**Comment by imranmaj** [comment]
Actually, now that #425 has been merged, I can try pulling the dev branch again.

**Pull request merged**